test/decorator_test.ts

```diff
@@ -52,4 +52,15 @@ describe('decorators', () => {
     expectTranslate('@IMPLEMENTS(Z) class X implements Y {}')
         .to.equal(' @ IMPLEMENTS ( Z ) class X implements Y , Z { }');
   });
+
+});
+
+describe('annotation/decorator hack', () => {
+  it('should strip "Annotation" from type names', () => {
+    expectTranslate('@CONST class FooAnnotation {}').to.equal(' @ CONST const class Foo { }');
+  });
+  it('should strip "Annotation" from imports', () => {
```

Comment:
Good observation, though outside of the parts that I'm overlooking - and I'm sure there are some - we should with this always strip "Annotation" from all type names, including super types. So This Should Not Happen (tm) (famous last words).